diff --git a/node_modules/@hotwired/turbo/dist/turbo.es2017-esm.js b/node_modules/@hotwired/turbo/dist/turbo.es2017-esm.js
index 086945c..b7710be 100644
--- a/node_modules/@hotwired/turbo/dist/turbo.es2017-esm.js
+++ b/node_modules/@hotwired/turbo/dist/turbo.es2017-esm.js
@@ -4620,6 +4620,7 @@ class History {
 class LinkPrefetchObserver {
   started = false
   #prefetchedLink = null
+  #pendingFetchRequests = new Map()
 
   constructor(delegate, eventTarget) {
     this.delegate = delegate;
@@ -4649,6 +4650,7 @@ class LinkPrefetchObserver {
     });
 
     this.eventTarget.removeEventListener("turbo:before-fetch-request", this.#tryToUsePrefetchedRequest, true);
+    this.eventTarget.removeEventListener("turbo:before-visit", this.#cancelPendingFetchRequests, true);
     this.started = false;
   }
 
@@ -4663,6 +4665,7 @@ class LinkPrefetchObserver {
     });
 
     this.eventTarget.addEventListener("turbo:before-fetch-request", this.#tryToUsePrefetchedRequest, true);
+    this.eventTarget.addEventListener("turbo:before-visit", this.#cancelPendingFetchRequests, true);
     this.started = true;
   }
 
@@ -4689,11 +4692,29 @@ class LinkPrefetchObserver {
 
         fetchRequest.fetchOptions.priority = "low";
 
+        // Memorize fetch request and cancel previous one to the same URL
+        const url = location.href;
+        const lastFetchRequest = this.#pendingFetchRequests.get(url);
+
+        if(lastFetchRequest) lastFetchRequest.cancel();
+        this.#pendingFetchRequests.set(url, fetchRequest);
+
         prefetchCache.putLater(location, fetchRequest, this.#cacheTtl);
       }
     }
   }
 
+  #cancelPendingFetchRequests = (event) => {
+    if (event.detail.fetchRequest) return
+
+    for (const [url, fetchRequest] of this.#pendingFetchRequests.entries()) {
+      if (url !== event.detail.url) {
+        fetchRequest.cancel();
+        this.#pendingFetchRequests.delete(url);
+      }
+    }
+  }
+
   #cancelRequestIfObsolete = (event) => {
     if (event.target === this.#prefetchedLink) this.#cancelPrefetchRequest();
   }
@@ -4737,7 +4758,13 @@ class LinkPrefetchObserver {
 
   requestErrored(fetchRequest) {}
 
-  requestFinished(fetchRequest) {}
+  requestFinished(fetchRequest) {
+    const url = fetchRequest.url.href;
+
+    if(this.#pendingFetchRequests.get(url) === fetchRequest) {
+      this.#pendingFetchRequests.delete(url);
+    }
+  }
 
   requestPreventedHandlingResponse(fetchRequest, fetchResponse) {}
 
diff --git a/node_modules/@hotwired/turbo/dist/turbo.es2017-umd.js b/node_modules/@hotwired/turbo/dist/turbo.es2017-umd.js
index 120ed11..1260129 100644
--- a/node_modules/@hotwired/turbo/dist/turbo.es2017-umd.js
+++ b/node_modules/@hotwired/turbo/dist/turbo.es2017-umd.js
@@ -4626,6 +4626,7 @@ Copyright © 2026 37signals LLC
   class LinkPrefetchObserver {
     started = false
     #prefetchedLink = null
+    #pendingFetchRequests = new Map()
 
     constructor(delegate, eventTarget) {
       this.delegate = delegate;
@@ -4655,6 +4656,7 @@ Copyright © 2026 37signals LLC
       });
 
       this.eventTarget.removeEventListener("turbo:before-fetch-request", this.#tryToUsePrefetchedRequest, true);
+      this.eventTarget.removeEventListener("turbo:before-visit", this.#cancelPendingFetchRequests, true);
       this.started = false;
     }
 
@@ -4669,6 +4671,7 @@ Copyright © 2026 37signals LLC
       });
 
       this.eventTarget.addEventListener("turbo:before-fetch-request", this.#tryToUsePrefetchedRequest, true);
+      this.eventTarget.addEventListener("turbo:before-visit", this.#cancelPendingFetchRequests, true);
       this.started = true;
     }
 
@@ -4695,11 +4698,29 @@ Copyright © 2026 37signals LLC
 
           fetchRequest.fetchOptions.priority = "low";
 
+          // Memorize fetch request and cancel previous one to the same URL
+          const url = location.href;
+          const lastFetchRequest = this.#pendingFetchRequests.get(url);
+
+          if(lastFetchRequest) lastFetchRequest.cancel();
+          this.#pendingFetchRequests.set(url, fetchRequest);
+
           prefetchCache.putLater(location, fetchRequest, this.#cacheTtl);
         }
       }
     }
 
+    #cancelPendingFetchRequests = (event) => {
+      if (event.detail.fetchRequest) return
+
+      for (const [url, fetchRequest] of this.#pendingFetchRequests.entries()) {
+        if (url !== event.detail.url) {
+          fetchRequest.cancel();
+          this.#pendingFetchRequests.delete(url);
+        }
+      }
+    }
+
     #cancelRequestIfObsolete = (event) => {
       if (event.target === this.#prefetchedLink) this.#cancelPrefetchRequest();
     }
@@ -4743,7 +4764,13 @@ Copyright © 2026 37signals LLC
 
     requestErrored(fetchRequest) {}
 
-    requestFinished(fetchRequest) {}
+    requestFinished(fetchRequest) {
+      const url = fetchRequest.url.href;
+
+      if(this.#pendingFetchRequests.get(url) === fetchRequest) {
+        this.#pendingFetchRequests.delete(url);
+      }
+    }
 
     requestPreventedHandlingResponse(fetchRequest, fetchResponse) {}
 
